src/App.cpp

```diff
@@ -573,6 +573,60 @@ void createAGVs()
                 }
                 socialForce->addAGV(agv);
 
+                int marker = numRunPerHallway * (juncIndexTemp + 1) - 1;
+                if ((int)inputData["runConcurrently"]["value"] == 1)
+                {
+                    marker = numRunPerHallway * 2 * (juncIndexTemp + 1) - 1;
+                }
+                if (agv->getId() == marker)
+                {
+                    juncIndexTemp = juncIndexTemp + 1;
+                    if (juncIndexTemp == juncDataList.size())
+                    {
+                        juncIndexTemp = 0;
+                    }
+                    hallwayLength = juncDataList[juncIndexTemp].items().begin().value();
+                    length1Side = (hallwayLength) / 2;
+                    juncDataTemp = {length1Side, length1Side};
+                }
+            }
+        }
+    }
+    else {

```

Comment:
runMode == 3

Reply:
I think it's not necessary because it has 3 cases: 0,1 or 3